### PR TITLE
Autohealing updates for path and status code range

### PR DESCRIPTION
**Pull request by puneetg1983**
With this PR, we are updating the AutoHeal UI to have this functionality
- Support for adding multiple rules for Slow Requests (Request Duration) based on request path
- Support for specifying a path for the Status Codes rules
- Support for specifying a range of HTTP status codes

image

Choose a single HTTP Status Code rule or a range of Status codes

image

image

### CODE REVIEW

File: AngularApp/projects/app-service-diagnostics/src/app/auto-healing/autohealing.component.ts [hunk 1]
```diff
@@ -309,6 +318,18 @@ export class AutohealingComponent implements OnInit {
     this.minProcessExecutionTimeExpanded = false;
   }
 
+  getStatusCodeRules() : StatusCodeRules {
+    if (this.autohealingSettings != null && this.autohealingSettings.autoHealRules != null && this.autohealingSettings.autoHealRules.triggers != null) {
+      return new StatusCodeRules(this.autohealingSettings.autoHealRules.triggers.statusCodes, this.autohealingSettings.autoHealRules.triggers.statusCodesRange);
+    }
+  }
+
+  getSlowRequestRules():SlowRequestsRules{
```
Comment (ShekharGupta1988) [inline AngularApp/projects/app-service-diagnostics/src/app/auto-healing/autohealing.component.ts:327]: nitpick : fix spacing

**Review by ShekharGupta1988 — APPROVED** [review]
Looks good. Can you just quickly run basic accessibility tests as well like keyboard accessibility, contrast ratio.

**Review by ShekharGupta1988 — COMMENTED** [review]
Looks good )
Can you also quickly check basic accessibility on the new option picker as well

File: AngularApp/projects/app-service-diagnostics/src/app/auto-healing/autohealing-config-summary/autohealing-config-summary.component.ts [hunk 2]
```diff
@@ -55,6 +55,19 @@ export class AutohealingConfigSummaryComponent implements OnInit, OnChanges {
         conditions.push(this.actualhealSettings.autoHealRules.triggers.slowRequests.count + ' requests take more than  ' + FormatHelper.timespanToSeconds(this.actualhealSettings.autoHealRules.triggers.slowRequests.timeTaken) + ' seconds in a duration of  ' + FormatHelper.timespanToSeconds(this.actualhealSettings.autoHealRules.triggers.slowRequests.timeInterval) + ' seconds');
       }
 
+      if (this.actualhealSettings.autoHealRules.triggers.slowRequestsWithPath != null) {
+        for (let index = 0; index < this.actualhealSettings.autoHealRules.triggers.slowRequestsWithPath.length; index++) {
+          const slowRequestRule = this.actualhealSettings.autoHealRules.triggers.slowRequestsWithPath[index];
+
+          let requestPath = "";
+          if (slowRequestRule.path) {
+            requestPath = " matching path '" + slowRequestRule.path + "' ";
```
[the remark this answers is not in the document]
Comment (puneetg1983) [reply AngularApp/projects/app-service-diagnostics/src/app/auto-healing/autohealing-config-summary/autohealing-config-summary.component.ts]: Yup for sure. I have used that syntax at a lot of places but I just ended up a bit lazy in this file and the whole code here is using string concatenation. changed this part though.

**Comment by puneetg1983** [comment]
> Looks good )
> Can you also quickly check basic accessibility on the new option picker as well

@ShekharGupta1988  - I lost the ability to run portal code locally. I will merge these changes and test in staging and raise a new PR if needed. 

BTW, it looks like colon on your keyboard is not working  

[ I am guessing you were trying to type :) instead of ) ]

**Pull request merged**